scripts/release.sh

```diff
@@ -166,9 +186,9 @@ function build_release() {
     mkdir "${dir}"
     pushd "${dir}"
 
-    green " - Building: ${os} ${arch} ${arm} with build tags '${buildtags}'"
-    env GOEXPERIMENT=loopvar CGO_ENABLED=0 GOOS="$os" GOARCH="$arch" GOARM=$arm GOAMD64="v1" go build -v -trimpath -ldflags="${ldflags}" -tags="${buildtags}" ${PKG}/cmd/tapd
-    env GOEXPERIMENT=loopvar CGO_ENABLED=0 GOOS="$os" GOARCH="$arch" GOARM=$arm GOAMD64="v1" go build -v -trimpath -ldflags="${ldflags}" -tags="${buildtags}" ${PKG}/cmd/tapcli
+    green " - Building: ${gotoolchain} ${os} ${arch} ${arm} with build tags '${buildtags}'"
```

Comment:
I'm very skeptical of using the `GOTOOLCHAIN` approach for this specific problem. Reproducibility is quite a fragile and difficult thing to achieve (took us many attempts on the `lnd` side to get the initial script right).
And any external mechanism (that is fairly new as well, only introduced in Go 1.21) will make it super hard to debug. The interaction with the docker based build and/or macOS also isn't clear to me.

Also, what's even the point of reproducing a build's signature if you don't have to do any manual check/verification and just blindly let the script do everything automatically?

So because I think this might introduce more problems than it solves, can't we just simply do a **check** for the right version and error out if it doesn't match?

Reply:
Latest commit is just a check which returns an error if the incorrect Go version is used.